modules/REST/V1/Languages.php

```diff
@@ -0,0 +1,865 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST\V1;
+
+use PLL_Language;
+use PLL_Translatable_Objects;
+use stdClass;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+use WP_Syntex\Polylang\Model\Languages as Languages_Model;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Languages REST controller.
+ *
+ * @since 3.7
+ *
+ * @phpstan-type LanguagePropsAll array{
+ *     term_id?: int,
+ *     slug?: non-empty-string,
+ *     locale?: non-empty-string,
+ *     name?: non-empty-string,
+ *     is_rtl?: bool,
+ *     term_group?: int,
+ *     flag_code?: non-empty-string,
+ *     no_default_cat?: bool
+ * }
+ * @phpstan-type LanguagePropsUpdate array{
+ *     term_id: int,
+ *     slug?: non-empty-string,
+ *     locale?: non-empty-string,
+ *     name?: non-empty-string,
+ *     is_rtl?: bool,
+ *     term_group?: int,
+ *     flag_code?: non-empty-string
+ * }
+ * @phpstan-type LanguagePropsCreate array{
+ *     locale: non-empty-string,
+ *     slug?: non-empty-string,
+ *     name?: non-empty-string,
+ *     is_rtl?: bool,
+ *     term_group?: int,
+ *     flag_code?: non-empty-string,
+ *     no_default_cat?: bool
+ * }
+ * @phpstan-type MandatoryPropsUpdate array{
+ *     lang_id: int,
+ *     locale: non-empty-string,
+ *     slug: non-empty-string,
+ *     name: non-empty-string,
+ *     rtl: bool,
+ *     term_group: int,
+ *     flag?: non-empty-string
+ * }
+ * @phpstan-type MandatoryPropsCreate array{
+ *    locale: non-empty-string,
+ *    slug: non-empty-string,
+ *    name: non-empty-string,
+ *    rtl: bool,
+ *    term_group: int,
+ *    flag: non-empty-string,
+ *    no_default_cat?: bool
+ * }
+ * @phpstan-type PreparedPropsAll object{
+ *     lang_id?: int,
+ *     name?: non-empty-string,
+ *     slug?: non-empty-string,
+ *     locale?: non-empty-string,
+ *     flag?: non-empty-string,
+ *     rtl?: bool,
+ *     term_group?: int,
+ *     no_default_cat?: bool
+ * }
+ */
```

Comment:
I'm surprised by that we suddenly need **all** these different types. In my previous review, it was much simpler. Does this increased complexity come from he introduction of `map_data_for_database()`?

Reply:
See my previous comment for some clarifications.